File: 백준/Silver/10828. 스택/스택.py


import sys
input = sys.stdin.readline
print = sys.stdout.write

N = int(input().strip())

def push(X,arr):
    arr.append(X)

def pop(arr):
    if arr:
        return arr.pop(-1)
    else:
        return -1

def size(arr):
    return len(arr)

def empty(arr):
    if len(arr) == 0:
        return 1
    else:
        return 0

def top(arr):
    if arr:
        return arr[-1]
    else:
        return -1

stack = []

for _ in range(N):
    command = input().strip().split()
    if command[0] == 'push':
        push(int(command[1]), stack)
    elif command[0] == 'pop':
        print(str(pop(stack)) + '\n')
    elif command[0] == 'size':
        print(str(size(stack)) + '\n')
    elif command[0] == 'empty':
        print(str(empty(stack)) + '\n')
    elif command[0] == 'top':
        print(str(top(stack)) + '\n')

File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0], 10);
const stack = [];
const result = [];

for (let i = 1; i <= n; i++) {
	const command = input[i].split(' ');

	switch (command[0]) {
		case 'push':
			stack.push(command[1]);
			break;

		case 'pop':
			result.push(stack.length !== 0 ? stack.pop() : -1);
			break;

		case 'size':
			result.push(stack.length);
			break;

		case 'empty':
			result.push(stack.length === 0 ? 1 : 0);
			break;

		case 'top':
			result.push(stack.length !== 0 ? stack[stack.length - 1] : -1);
			break;
	}
}

console.log(result.join('\n'));
